addons/binding/org.openhab.binding.tadoac/src/main/java/org/openhab/binding/tadoac/handler/TadoACHandler.java

```diff
@@ -0,0 +1,224 @@
+            if (username == null || password == null) {
+                logger.error("login credentials not configured");
+                updateStatus(ThingStatus.OFFLINE);
+                return;
+            }
+            TadoACConnector connector = new TadoACConnector(username, password);
+            TadoACSetting newSetting = connector.getSetting(homeid, zoneid);
+
+            // Check each part and copy them to the previous state (No loss of data)
+            TadoACPower newPower = newSetting.getPower();
+            if (newPower != null) {
+                setting.setPower(newPower);
+                updateState(CHANNEL_POWER, newPower == TadoACPower.ON ? OnOffType.ON : OnOffType.OFF);
+            }
+            TadoACFanSpeed newFanSpeed = newSetting.getFanSpeed();
+            if (newFanSpeed != null) {
+                setting.setFanSpeed(newFanSpeed);
+                int fanSpeed = 1;
+                switch (newFanSpeed) {
+                    case LOW:
+                        fanSpeed = 1;
+                        break;
+                    case MIDDLE:
+                        fanSpeed = 2;
+                        break;
+                    case HIGH:
+                        fanSpeed = 3;
+                        break;
+                    default:
+                        break;
+                }
+                updateState(CHANNEL_FANSPEED, new DecimalType(fanSpeed));
+            }
+            TadoACMode newMode = newSetting.getMode();
+            if (newMode != null) {
+                setting.setMode(newMode);
+                int mode = 1;
+                switch (newMode) {
+                    case COOL:
+                        mode = 1;
+                        break;
+                    case DRY:
+                        mode = 2;
+                        break;
+                    case FAN:
+                        mode = 3;
+                        break;
+                    default:
+                        break;
+                }
+                updateState(CHANNEL_MODE, new DecimalType(mode));
+            }
+            TadoACTemperature newTemp = newSetting.getTemperature();
+            if (newTemp != null) {
+                setting.setTemperature(newTemp);
+                updateState(CHANNEL_TEMP, new DecimalType(newTemp.getCelsius()));
+            }
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            logger.debug("error while polling for new state", e);
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        setting = new TadoACSetting();
+        int interval = ((BigDecimal) getConfig().get("interval")).intValue();
+        if (interval < 30) {
+            logger.info("Minimum refresh interval is 30 seconds. We do not want to annoy Tado");
+            interval = 30;
+        }
+        scheduler.scheduleAtFixedRate(() -> refreshSetting(), 15, interval, TimeUnit.SECONDS);
```

Comment:
Please use [scheduleWithFixedDelay](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)) which is preferred over scheduleAtFixedRate, see [Coding Guidelines D.3](http://docs.openhab.org/developers/development/guidelines.html#d-runtime-behavior)